components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/dao/AccessTokenDAOImpl.java

```diff
@@ -903,6 +924,9 @@ public AccessTokenDO getAccessToken(String accessTokenIdentifier, boolean includ
 
             prepStmt.setString(1,
                     getHashingPersistenceProcessor().getProcessedAccessTokenIdentifier(accessTokenIdentifier));
+            if (isTenantQualifiedUrlsEnabled && tenantDomain != null) {
+                prepStmt.setInt(2, IdentityTenantUtil.getTenantId(tenantDomain));
```

Comment:
If the tenant domain is only used to derive the tenantId, shall we do it in the first step and continue rather than validating the tenant domain in each step?

Reply:
Removed from latest commits